redfish-core/lib/log_services.hpp

```diff
@@ -3411,6 +3436,610 @@ inline void getSystemHardwareIsolationLogService(
+ *
+ * @note - This function will get the associated redfish uri prefix of "Id"
+ *         ("<str>") and look at the predefined redfish list (which will
+ *         contains the pair <DBus Interfaces (which represents redfish
+ *         resource), RedfishUri> of elements) to get the dbus interfaces and
+ *         that is used to get the dbus objects which will compare with the
+ *         given dbus object path to fill the respective uri Id in the
+ *         given redfishUri.
+ *       - This function will run recursively until filling all Ids in given
+ *         given redfishUri.
+ */
+inline void
+    fillRedfishUriWithIds(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                          const size_t entryJsonIdx,
+                          const sdbusplus::message::object_path& dbusObjPath,
+                          std::string& redfishUri)
+{
+
+    std::string uriIdPattern{"<str>"};
+    if (auto uriIdPos = redfishUri.rfind(uriIdPattern);
+        uriIdPos != std::string::npos)
+    {
+        // Reduce by "1" from uriIdPos to get the redfish uri of uriIdPattern
+        // For example,
+        // "/redfish/v1/Systems/system/Processors/<str>/SubProcessors/core0"
+        //                                       ^
+        auto parentRedfishUri = redfishUri.substr(0, uriIdPos - 1);
+        auto parentRedfishUriIt =
+            std::find_if(redfishUriList.begin(), redfishUriList.end(),
+                         [parentRedfishUri](const auto& ele) {
+                             return parentRedfishUri == std::get<1>(ele);
+                         });
+
+        if (parentRedfishUriIt == redfishUriList.end())
+        {
+            BMCWEB_LOG_ERROR
+                << "Failed to get parent Redfish URI DBus interface for the "
+                   "identified Redfish URI: "
+                << redfishUri
+                << " of the given DBus object path: " << dbusObjPath.str;
+            return;
+        }
+
+        crow::connections::systemBus->async_method_call(
+            [asyncResp, entryJsonIdx, dbusObjPath, uriIdPos, uriIdPattern,
+             parentRedfishUri,
+             redfishUri](const boost::system::error_code ec,
+                         const std::vector<std::string>& objects) mutable {
+                if (ec || objects.empty())
+                {
+                    BMCWEB_LOG_ERROR << "Failed to get parent DBus path for "
+                                        "identified parent Redfish URI: "
+                                     << parentRedfishUri
+                                     << " of the given DBus object path: "
+                                     << dbusObjPath.str;
+                    return;
+                }
+
+                std::string uriIdObj;
+                for (auto& object : objects)
+                {
+                    if (dbusObjPath.str.find(object) == std::string::npos)
+                    {
+                        continue;
```

Comment:
Instead of using continue, you can have the != check and break once the object is found